fms/models/hf/utils.py

```diff
@@ -130,3 +135,68 @@ def to_hf_api(model: nn.Module, **override_config_kwargs) -> "HFModelArchitectur
 
     hf_adapted_cls = _fms_to_hf_adapt_map[model_type]
     return hf_adapted_cls.from_fms_model(model, **override_config_kwargs)
+
+
+def get_model(
+    model_id_or_path: Union[str, os.PathLike], device: Union[str, torch.device] = "cpu"
+) -> nn.Module:
+    """
+    get an FMS model from a huggingface checkpoint
+
+    Parameters
+    ----------
+    model_id_or_path: Union[str, os.PathLike]
+        The huggingface hub model id or a local path. If the local path exists, the model will be loaded directly from
+        the local path, otherwise the huggingface cache will be checked. If the huggingface cache does not contain the
+        model, then the weights will be downloaded and stored into the huggingface cache
+    device: Union[str, torch.device]
+        the device to load the model weights to
+
+    Returns
+    -------
+    nn.Module
+        an fms equivalent implementation of an HF model
+    """
+    if not os.path.exists(model_id_or_path):
+        model_id_or_path = snapshot_download(repo_id=model_id_or_path)
```

Comment:
is there a use-case for an option to not fetching the weights, i.e. would we ever want to pretrain an HF configuration of an fms model? I guess in that case we'd probably just add the fms config

Reply:
Ya not sure if that is a use case we need to support, I would just use `get_model` directly then